src/charm.py

```diff
@@ -93,32 +93,11 @@ def __init__(self, *args):
     # BEGIN: hook functions
     def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         """Listen to changes in the application configuration."""
-        previous_config = self.expose_external
-        external_config = self.model.config["expose-external"]
-        if external_config not in Config.ExternalConnections.VALID_EXTERNAL_CONFIG:
-            logger.error(
-                "External configuration: %s for expose-external is not valid, should be one of: %s",
-                external_config,
-                Config.ExternalConnections.VALID_EXTERNAL_CONFIG,
-            )
-            self.status.set_and_share_status(Config.Status.INVALID_EXTERNAL_CONFIG)
+        if not self.is_user_external_config_valid():
+            self.set_status_invalid_external_config()
             return
 
-        self.expose_external = external_config
-        if external_config == Config.ExternalConnections.EXTERNAL_NODEPORT:
-            # every unit attempts to create a nodeport service - if exists, will silently continue
-            self.node_port_manager.apply_service(
-                service=self.node_port_manager.build_node_port_services(
-                    port=Config.MONGOS_PORT
-                )
-            )
-
-        if (
-            external_config == Config.ExternalConnections.NONE
-            and previous_config == Config.ExternalConnections.EXTERNAL_NODEPORT
-        ):
-            # TODO DPE-5268 - support revoking external access
-            pass
+        self.update_external_services()
```

Comment:
This eventually raises 2 potential exception: 1 when the service metadata is not available and the other one when the service fails. How should we handle these 2?

Reply:
1. If metadata is not available - retrying will prob not do anything useful. I believe it is best to let it error out
2. Same goes for service failing. 

If you think otherwise I'd love to hear the reasoning and expand the cases.